pkg/handler/project/project.go

```diff
@@ -903,3 +903,121 @@ func (h *handler) Details(c *gin.Context) {
 
 	c.JSON(http.StatusOK, view.CreateResponse(view.ToProjectData(project), nil, nil, nil, ""))
 }
+
+// UpdateGeneralInfo godoc
+// @Summary Update general info of the project by id
+// @Description Update general info of the project by id
+// @Tags Project
+// @Accept  json
+// @Produce  json
+// @Param Authorization header string true "jwt token"
+// @Param id path string true "Project ID"
+// @Param Body body UpdateGeneralInfoInput true "Body"
+// @Success 200 {object} view.UpdateProjectGeneralInfoResponse
+// @Failure 400 {object} view.ErrorResponse
+// @Failure 404 {object} view.ErrorResponse
+// @Failure 500 {object} view.ErrorResponse
+// @Router /projects/{id}/general-info [put]
+func (h *handler) UpdateGeneralInfo(c *gin.Context) {
+	projectID := c.Param("id")
+
+	var body UpdateGeneralInfoInput
+	if err := c.ShouldBindJSON(&body); err != nil {
+		if err != nil {
+			c.JSON(http.StatusBadRequest, view.CreateResponse[any](nil, nil, err, body, ""))
+			return
+		}
+	}
+
+	// TODO: can we move this to middleware ?
+	l := h.logger.Fields(logger.Fields{
+		"handler": "project",
+		"method":  "UpdateGeneralInfo",
+		"request": body,
+	})
+
+	// Check project exists
+	check, err := h.store.Project.Exists(h.repo.DB(), projectID)
+	if err != nil {
+		l.Error(err, "error check existence of project")
+		c.JSON(http.StatusInternalServerError, view.CreateResponse[any](nil, nil, err, body, ""))
+		return
+	}
+
+	if !check {
+		l.Error(err, "error project not found")
+		c.JSON(http.StatusInternalServerError, view.CreateResponse[any](nil, nil, ErrProjectNotFound, body, ""))
+		return
+	}
+
+	// Check country exists
+	check, err = h.store.Country.Exists(h.repo.DB(), body.CountryID.String())
+	if err != nil {
+		l.Error(err, "error check existence of country")
+		c.JSON(http.StatusInternalServerError, view.CreateResponse[any](nil, nil, err, body, ""))
+		return
+	}
+
+	if !check {
+		l.Error(err, "error country not found")
+		c.JSON(http.StatusInternalServerError, view.CreateResponse[any](nil, nil, ErrCountryNotFound, body, ""))
+		return
+	}
+
+	// Check valid stack id
+	stacks, err := h.store.Stack.All(h.repo.DB())
+	if err != nil {
+		l.Error(err, "error when finding stacks")
+		c.JSON(http.StatusInternalServerError, view.CreateResponse[any](nil, nil, err, body, ""))
+		return
+	}
+
+	stackMap := model.ToStackMap(stacks)
+	for _, sID := range body.Stacks {
+		_, ok := stackMap[sID]
+		if !ok {
+			l.Error(errStackNotFound(sID.String()), "error stack not found")
```

Comment:
`error` in the error message doesn't make sense (same as with all error message)